File: cpp File/5568/5568.cpp
#include <iostream>
#include <string>
#include <set>

using namespace std;

int n, k, c, ans;
int card[10];
bool visited[10];

set<string> info;

void solve(int depth, string str)
{
	if (depth == k)
	{
		if (info.count(str) == 0)
		{
			info.insert(str);
			ans++;
		}

		return;
	}

	for (int i = 0; i < n; i++)
	{
		if (visited[i] == true) continue;

		visited[i] = true;
		solve(depth + 1, str + to_string(card[i]));
		visited[i] = false;
	}
}

int main()
{
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	cin >> n;
	cin >> k;

	for (int i = 0; i < n; i++)
	{
		cin >> card[i];
	}

	solve(0, "");

	cout << ans << "\n";

	return 0;
}

File: kotlin File/5568.kt
var n = 0
var k = 0
var ans = 0

val card = Array(10) { 0 }
val visited = Array(10) { false }
val hset = hashSetOf<String>()

fun main() {
    n = readLine()!!.toInt()
    k = readLine()!!.toInt()

    for (i in 0 until n) {
        card[i] = readLine()!!.toInt()
    }

    DFS(0, "")

    println(ans)
}

fun DFS(depth: Int, str: String) {
    if (depth == k) {
        if (!hset.contains(str)) {
            hset.add(str)
            ans++
        }

        return
    }

    for (i in 0 until n) {
        if (visited[i]) continue

        visited[i] = true
        DFS(depth + 1, str + card[i].toString())
        visited[i] = false
    }
}